### ISSUE TITLE
MessageBar A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#7762]
Tracking all of the issues brought up with the 1/22 accessibility pass on MessageBar control 
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B8f6b9510-2d32-45ba-bc39-37aa879c7eba%7D&action=default&uid=%7B8F6B9510-2D32-45BA-BC39-37AA879C7EBA%7D&ListItemId=1676&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble) 
  
 - [x] Bug1: Bug1: [Accessibility][Visual requirements]: Luminosity ratio of link 'Visit our website' under 'Error, single line, with dismiss button - custom styles' section is less than 4.5:1 ratio.  
 - [x] Bug2: [Accessibility][Keyboard navigation]: Focus indicator is not properly visible on  'visit our website' link. 
 - [x] Bug3: [Accessibility][Screen Reader]: Narrator in scan  mode doesn't announce the role of  'See More' , 'Close ('X')' buttons and 'visit our website' link  under 'Blocked, single line, with dismiss button and truncated text - custom styles' section. " 
 - [x] Bug4: [Accessibility][Screen Reader]: Narrator doesn't announce the state of 'See More' button while navigating with tab key. 
 - [x] Bug5: [Accessibility][Keyboard Navigation]: Text is disappearing from the screen when keyboard focus moves to the 'See more' button. " 
- [x] Bug6: [Accessibility][ Supporting the platform]: Controls present in the ‘MessageBar’ are not adopting high contrast settings. " 
- [x] Bug7: [Accessibility][Suggestion]: Appeared message bar should have ‘Aria-live’ property and region landmark. " 
  
 Component: MessageBar  
@micahgodbolt

**Comment by Hardhik** [comment]
Please add the following issue on Message w.r.t. Screen Reader
When we navigate to the MessageBar while using Narrator with Caps Key + Arrow, it announces as "Status" and we have to click Caps Key + Arrow once again to read the text.

For example
Narrator reads the following for the example in https://developer.microsoft.com/en-us/fabric#/controls/web/messagebar 

`Show code, button, 
Info/Default MessageBar, 

**status**, 

Info lorem ipsum dolor sit amet, a elit sem interdum consectetur adipiscing elit., 

link, Visit our website., 

Error MessageBar - single line, with dismiss button, 

<say-as interpret-as="spell">l</say-as>

Search or enter web address, edit, https://developer.microsoft.com/en-us/fabric#/controls/web/messagebar, navigation landmark,`

The reading out of "status" is not clear.
It should announce the text present on the page on the first caps+arrow key, the word "Status" should not be announced.

**Comment by micahgodbolt** [comment]
2 - Seems to be working now. 
![image](https://user-images.githubusercontent.com/1434956/65469812-96630380-de1d-11e9-9078-03202359f9a1.png)

**Comment by micahgodbolt** [comment]
6 - high contrast is working fine 

![image](https://user-images.githubusercontent.com/1434956/65905134-d32a7f80-e374-11e9-8003-5567830bf30b.png)


![image](https://user-images.githubusercontent.com/1434956/65905147-d9b8f700-e374-11e9-9b36-142da0f70f75.png)

![image](https://user-images.githubusercontent.com/1434956/65905164-e3daf580-e374-11e9-9c78-31ff72a69881.png)

**Comment by micahgodbolt** [comment]
Looks like everything except the color contrast issue is resolved. by my upcoming PR. The color will get broken out into a different issue.

**Comment by micahgodbolt** [comment]
Removed the custom styles section as it was full of a11y issues and didn't serve any specific purpose.
